File: backend/app/routers/intelligence.py
from fastapi import APIRouter, Depends
from sqlalchemy.orm import Session

from app.core.database import get_db
from app.schemas.intelligence import (
    InventoryIntelligenceResponse,
    ProductProfitabilityResponse,
    ProfitSummaryResponse,
)
from app.services.profit_intelligence import (
    get_inventory_intelligence,
    get_product_profitability,
    get_profit_summary,
)


router = APIRouter(
    prefix="/api/v1/intelligence",
    tags=["Intelligence"],
)


@router.get(
    "/profit-summary",
    response_model=ProfitSummaryResponse,
)
def profit_summary(
    db: Session = Depends(get_db),
):
    return get_profit_summary(db)


@router.get(
    "/product-profitability",
    response_model=list[ProductProfitabilityResponse],
)
def product_profitability(
    db: Session = Depends(get_db),
):
    return get_product_profitability(db)


@router.get(
    "/inventory",
    response_model=InventoryIntelligenceResponse,
)
def inventory_intelligence(
    db: Session = Depends(get_db),
):
    return get_inventory_intelligence(db)

File: frontend/src/services/intelligence.ts
import { apiFetch } from "../lib/api";
import type {
  InventoryIntelligence,
  ProductProfitability,
  ProfitSummary,
  RevenueTrend,
} from "../types/intelligence";

export function getProfitSummary() {
  return apiFetch<ProfitSummary>(
    "/api/v1/intelligence/profit-summary",
  );
}

export function getProductProfitability() {
  return apiFetch<ProductProfitability[]>(
    "/api/v1/intelligence/product-profitability",
  );
}

export function getInventoryIntelligence() {
  return apiFetch<InventoryIntelligence>(
    "/api/v1/intelligence/inventory",
  );
}

export function getRevenueTrend() {
  return apiFetch<RevenueTrend[]>(
    "/api/v1/intelligence/revenue-trend",
  );
}